lib/async/enumerable/fiber_limiter.rb

```diff
@@ -35,7 +35,7 @@ def with_bounded_concurrency(early_termination: false, &block)
       # Gets max fibers (instance or global default).
       # @return [Integer] Maximum concurrent fibers
       def max_fibers
-        @max_fibers || Async::Enumerable.max_fibers
+        __async_enumerable_config&.max_fibers || Async::Enumerable.max_fibers
```

Comment:
let's make `__async_enumerable_config` do the or logic between the instance config and the Async::Enumerable config. then, this accessor can just call `__async_enumerable_config.max_fibers` and it doesn't need to know about the structure of the config system.